exposed-java-time/src/test/kotlin/org/jetbrains/exposed/DateLiteralTest.kt

```diff
@@ -0,0 +1,39 @@
+package org.jetbrains.exposed
+
+import junit.framework.TestCase.assertNull
+import org.jetbrains.exposed.DateLiteralTest.TableWithDate.date
+import org.jetbrains.exposed.dao.id.IntIdTable
+import org.jetbrains.exposed.sql.insert
+import org.jetbrains.exposed.sql.javatime.date
+import org.jetbrains.exposed.sql.javatime.dateLiteral
+import org.jetbrains.exposed.sql.selectAll
+import org.jetbrains.exposed.sql.tests.DatabaseTestsBase
+import org.jetbrains.exposed.sql.tests.shared.assertEquals
+import org.junit.Test
+import java.time.LocalDate
+
+class DateLiteralTest : DatabaseTestsBase() {
```

Comment:
If creating a new test file, please name it something more general like `DateTimeLiteralTest`, so that we can do more in it later.

More than likely `timestampLiteral()` and `timeLiteral()` will also not work in Oracle as I don't believe that they are tested anywhere else. So we can set this up to test all literals eventually in other PRs.

Reply:
You are right. I found same issues with datetime, timestamp and time literals. In general, I fixed these cases, but only with dateLiteral I got weird behaviour. 

I created the test that should work, but it doesn't. I marked it as @Ignore. It should work in the same way as timestamp test, because it generated almost the same sql, but the query cannot find the entry after insert. If I log SQL commands and run them manually in DB (create table, insert, and query), I receive the right result. At the current moment, no idea why it doesn't work...